default/AI/ProductionAI.py

```diff
@@ -66,22 +101,37 @@ def update_best_mil_ship_rating():
     cost = best_design.productionCost(fo.empireID(), build_choices[0])
     bestMilRatingsHistory[fo.currentTurn()] = (stats['overall'], cost)
 
+
 def cur_best_mil_ship_rating():
-    if fo.currentTurn() not in bestMilRatingsHistory:
-        update_best_mil_ship_rating()
-    return bestMilRatingsHistory[ fo.currentTurn()][0]
+    priority = EnumsAI.AIPriorityType.PRIORITY_PRODUCTION_MILITARY
+    if priority in design_cache:  # use new framework
+        try:
+            return design_cache[priority][0][0]
+        except Exception:
+            print traceback.format_exc()
+            return 0.0001
+    else:
+        if fo.currentTurn() not in bestMilRatingsHistory:
+            update_best_mil_ship_rating()
```

Comment:
```bestMilRatingsHistory``` is a cache used in the old implementation and is not used in the new framework anymore. If I haven't overlooked something, that particular cache is only used in the functions I adjusted for the new framework, anyway.
The new framework uses ```design_cache``` instead which is much more general and powerful.

Currently, both old and new implementation exist in parallel (and are both used by every other AI respectively).